build/post_build.sh

```diff
@@ -1,22 +1,15 @@
 #!/bin/bash
 echo "==========STARTING POST_BUILD=========="
 
-for BUILDER in ${SPEL_BUILDERS//,/ }; do
-  AMI_NAME="validation-$SPEL_IDENTIFIER-$BUILDER-$SPEL_VERSION.x86_64-gp2"
-  # shellcheck disable=SC2030,SC2031
-  AMI_ID=$(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 describe-images --filters Name=name,Values="$AMI_NAME" --query 'Images[0].ImageId' --out text --profile "$SPEL_IDENTIFIER")
-  echo "Trying to deregister: $AMI_NAME:$AMI_ID in $AWS_REGION"
-  # shellcheck disable=SC2030,SC2031,SC2091
-  $(export AWS_DEFAULT_REGION="$AWS_REGION"; test "$AMI_ID" != "None" && aws ec2 deregister-image --image-id "$AMI_ID" --profile "$SPEL_IDENTIFIER")
+if [ "$SPEL_CI" = "true" ]; then
+    for BUILDER in ${SPEL_BUILDERS//,/ }; do
+        AMI_NAME="$SPEL_IDENTIFIER-$BUILDER-$SPEL_VERSION.x86_64-gp2"
+        # shellcheck disable=SC2030,SC2031
+        AMI_ID=$(export AWS_DEFAULT_REGION="$AWS_REGION"; aws ec2 describe-images --filters Name=name,Values="$AMI_NAME" --query 'Images[0].ImageId' --out text --profile "$SPEL_IDENTIFIER")
```

Comment:
Similarly, why this rather than:

~~~
AMI_ID=$(export AWS_DEFAULT_REGION="$AWS_REGION" ;
   aws ec2 describe-images --owner self --query \
     'Images[?Name == `'"${SPEL_IDENTIFIER-$BUILDER-$SPEL_VERSION}.x86_64-gp2"'`].ImageId' \
     --profile "${SPEL_IDENTIFIER}"
)
~~~

Reply:
Sure, can always cleanup old code if it helps. Though it's usually desirable to `--filter` these describe calls to limit the data handled by the `--query`, otherwise it's just a lot of wasted I/O and memory.